api/ttn/lorawan/v3/gateway.proto

```diff
@@ -249,6 +251,272 @@ message Gateways {
+      ipv4: true
+      ignore_empty: true
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      type: STRING,
+      format: "string",
+      example: "\"192.0.2.0\""
+    }
+  ];
+  repeated string dns_servers = 4 [
+    (validate.rules).repeated = {
+      max_items: 2,
+      items: {
+        string: {ipv4: true}
+      }
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      type: STRING,
+      format: "string",
+      example: "[\"192.0.2.0\",\"192.0.2.1\"]"
+    }
+  ];
+}
+
+message ManagedGatewaySystemStatus {
+  // CPU temperature (Celcius).
+  google.protobuf.FloatValue cpu_temperature = 1;
+}
+
+message ManagedGatewayControllerConnection {
+  ManagedGatewayNetworkInterfaceType network_interface_type = 1 [(validate.rules).enum.defined_only = true];
+}
+
+message ManagedGatewayGatewayServerConnection {
+  ManagedGatewayNetworkInterfaceType network_interface_type = 1 [(validate.rules).enum.defined_only = true];
+  string address = 2;
+}
+
+message ManagedGatewayNetworkInterfaceInfo {
+  ManagedGatewayNetworkInterfaceStatus status = 1 [(validate.rules).enum.defined_only = true];
+  bool dhcp_enabled = 2;
+  ManagedGatewayNetworkInterfaceAddresses addresses = 3;
+}
+
+message ManagedGatewayCellularBackhaul {
+  ManagedGatewayNetworkInterfaceInfo network_interface = 1;
+  string operator = 2 [(validate.rules).string.max_len = 32];
+  google.protobuf.FloatValue rssi = 3;
+}
+
+message ManagedGatewayWiFiBackhaul {
+  ManagedGatewayNetworkInterfaceInfo network_interface = 1;
+  string ssid = 2 [(validate.rules).string.max_len = 32];
+  bytes bssid = 3 [
+    (validate.rules).bytes = {len: 6},
+    (thethings.json.field) = {
+      marshaler_func: "go.thethings.network/lorawan-stack/v3/pkg/types.MarshalHEXBytes",
+      unmarshaler_func: "go.thethings.network/lorawan-stack/v3/pkg/types.Unmarshal6Bytes"
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      type: STRING,
+      format: "string",
+      example: "\"EC656E000100\""
+    }
+  ];
+  uint32 channel = 4 [(validate.rules).uint32 = {gte: 1}];
+  string authentication_mode = 5;
+  google.protobuf.FloatValue rssi = 6;
+}
+
+message ManagedGatewayEthernetBackhaul {
+  ManagedGatewayNetworkInterfaceInfo network_interface = 1;
+}
+
+message ManagedGatewayEventData {
+  oneof data {
+    ManagedGateway entity = 1;
+    Location location = 2;
+    ManagedGatewaySystemStatus system_status = 3;
+    ManagedGatewayControllerConnection controller_connection = 4;
+    ManagedGatewayGatewayServerConnection gateway_server_connection = 5;
+    ManagedGatewayCellularBackhaul cellular_backhaul = 6;
+    ManagedGatewayWiFiBackhaul wifi_backhaul = 7;
+    ManagedGatewayEthernetBackhaul ethernet_backhaul = 8;
+  }
+}
+
+message ManagedGatewayWiFiProfile {
+  option (thethings.flags.message) = {
+    select: true,
+    set: true
+  };
+  string profile_id = 1 [(validate.rules).string = {
```

Comment:
Should this be a first class identifier (similar to `AlertProfileIdentifiers`)?

Reply:
Hmm I think it's something opaque for TTS, it's an external ID, not a TTS concept. I'm a bit divided to do this, because then we'll have to define messages for potentially many external IDs.

Reply:
Coming back on this, let's keep this a string as it is a TTS-external identifier.